tools/apiview/emitters/cadl-apiview/src/emitter.ts

```diff
@@ -58,10 +58,11 @@ function createApiViewEmitter(program: Program, options: ResolvedApiViewEmitterO
     apiview.emit(program);
     apiview.resolveMissingTypeReferences();
 
-    const tokenJson = JSON.stringify(apiview.asApiViewDocument()) + "\n";
-    await emitFile(program, {
-      path: options["output-file"],
-      content: tokenJson,
-    });
+    if (!program.compilerOptions.noEmit && !program.hasError()) {
```

Comment:
just adding the same comment here as on Izzy PR for tracking, this is ok for now until we figure out if we want to change the behavior of `noEmit` but this should be change in the future to either respect the current meaning of noEmit(run your emitter but don't write files) or that noemit will not even trigger the emitter making this check obsolete